examples/cpp/flow_control/PublisherApp.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file PublisherApp.hpp
+ *
+ */
+
+
+#ifndef FASTDDS_FLOW_CONTROL_PUBLISHER_APP_HPP_
+#define FASTDDS_FLOW_CONTROL_PUBLISHER_APP_HPP_
+
+#include <condition_variable>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+
+#include "Application.hpp"
+#include "CLIParser.hpp"
+#include "FlowControlPubSubTypes.h"
+
+using namespace eprosima::fastdds::dds;
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace flow_control {
+
+class PublisherApp : public Application, public DataWriterListener
+{
+public:
+
+    PublisherApp(
+            const CLIParser::flow_control_config& config);
+
+    ~PublisherApp();
+
+    //! Publisher matched method
+    void on_publication_matched(
+            DataWriter* writer,
+            const PublicationMatchedStatus& info) override;
+
+    //! Run publisher
+    void run() override;
+
+    //! Stop publisher
+    void stop() override;
+
+private:
+
+    //! Return the current state of execution
+    bool is_stopped();
+
+    //! Publish a sample
+    bool publish(
+            DataWriter* writer_,
+            int& msgsent,
+            FlowControl msg);
```

Comment:
Following the user data approach, you don't need the data as input, you could set it internally